airbyte-server/src/main/java/io/airbyte/server/converters/ConfigConverter.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.server.converters;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.yaml.Yamls;
+import io.airbyte.config.ConfigSchema;
+import io.airbyte.validation.json.JsonSchemaValidator;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConfigConverter {
```

Comment:
it was pretty hard for me to understand what this class was doing. I think it's largely a naming thing that's making it hard for me. 

If I were try to explain to someone what this class does, I'd say, "It hands writing data to a file archive _and_ reading data from the file archive."

Based on that, maybe we call this class `ConfigFileArchive`. and then `writeConfigList` => `writeConfigsToArchive`, and `readConfigList` => `readConfigsFromArchive`. Feel free to riff on this and find what makes sense to you, but I think the current names are really hard to penetrate.

Reply:
In addition I would make sure you had javadoc style comments, either for the overall class or for the two public methods that explain what is happening and that make a note that all of the configs will be validated!

Reply:
Sure, but should this class still live in the converters package?

I initially thought about it like "it converts a Configuration File into an Archived Configuration File"

Reply:
i think that's fine for now.